lib/proxy/mutators/urlencoded_bodyparser.js

```diff
@@ -17,58 +17,19 @@ module.exports = function(proxy) {
-    //   proxy.logger.warn(module_tag, 'Stashed original "Content-Type": "%s"', req.original_content_type);
-    // }
-
-    // If the headers indicate that the content is not UTF-8, then convert the contents to UTF-8.
-    // var test = /^.*?charset\s*=([A-z0-9-_]+)\s*(;|\s)?/i;
-    // var original_encoding = (req.original_content_encoding || req.original_content_type || '').replace(test, '$1');
-    // if (original_encoding !== '' && iconv.encodingExists(original_encoding)) {
-    //   if (req.url !== '') {
-    //     var original_url = req.url;
-    //     req.url = iconv.decode(original_url, original_encoding);
-    //     proxy.logger.debug(module_tag, 'req.url: (%s) %s changed to (UTF-8) %s', original_encoding, util.inspect(original_url), util.inspect(req.url));
-    //   }
-    //   if (typeof req.body !== 'undefined' && req.body !== '') {
-    //     var original_body = req.body;
-    //     req.body = iconv.decode(original_body, original_encoding);
-    //     proxy.logger.debug(module_tag, 'req.body: (%s) %s changed to (UTF-8) %s', original_encoding, util.inspect(original_body), util.inspect(req.body));
-    //   }
-    // } else {
-    //   proxy.logger.error(module_tag, 'Interpreted charset "%s" is invalid', original_encoding);
-    // }
+    if (req.headers['content-type'] && req.headers['content-type'].indexOf('application/x-www-form-urlencoded') === 0) {
+      req.cached_header = req.headers['content-type'];
+      req.headers['content-type'] = 'application/x-www-form-urlencoded';
+    }
 
     var parser = body_parser.urlencoded({ extended: true, limit: '1mb' });
     parser(req, res, function(err) {
       // Handle errors
-      if (err && err.status === 413) {
-        return next(err);
-      } else if (err && err.status === 415) {
-        proxy.logger.error(module_tag, 'Failed to handle request due to non-"utf-8" charset: req.headers', err, util.inspect(req.headers));
-        return next(err);
-      } else if (err) {
-        proxy.logger.error(module_tag, 'Failed to parse body: %s', err);
-        return next(err);
-      }
+      if (err) return next(err);
 
-      // Restore stashed headers for use by the next middleware.
-      // if (typeof req.original_content_encoding !== 'undefined' && req.original_content_encoding !== '') {
-      //   req.headers['Content-Encoding'] = req.original_content_encoding;
-      //   proxy.logger.warn(module_tag, 'Restored request header "Content-Encoding": "%s", which was temporarily removed', req.original_content_encoding);
-      // }
-      // if (typeof req.original_content_type !== 'undefined' && req.original_content_type !== '') {
-      //   req.headers['Content-Type'] = req.original_content_type;
-      //   proxy.logger.warn(module_tag, 'Restored request header "Content-Type": "%s", which was temporarily removed', req.original_content_type);
-      // }
+      if (req.cached_header) {
+        req.headers['content-type'] == req.cached_header;
```

Comment:
There's a `==` instead of a `=`, which means that this line has no effect. 

I'm seeing build errors if the header is re-assigned to it's original value. I'm looking for other locations where body-parser is used. It's a dependency of express and connect as well